test/parallel/test-typescript.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+const { spawnPromisified } = require('../common');
+const fixtures = require('../common/fixtures');
+const { deepStrictEqual, match, strictEqual } = require('node:assert');
+const { test } = require('node:test');
+
+test('execute a typescript file', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-typescript.ts'),
+  ]);
+
+  deepStrictEqual(result, {
+    code: 0,
+    signal: null,
+    stderr: '',
+    stdout: 'Hello, TypeScript!\n',
+  });
+});
+
+test('execute a typescript file with imports', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/b.ts'),
+  ]);
+
+  deepStrictEqual(result, {
+    code: 0,
+    signal: null,
+    stderr: '',
+    stdout: 'Hello, TypeScript!\n',
+  });
+});
+
+test('execute a typescript with node_modules', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-typescript-node-modules.ts'),
+  ]);
+
+  deepStrictEqual(result, {
+    code: 0,
+    signal: null,
+    stderr: '',
+    stdout: 'Hello, TypeScript!\n',
+  });
+});
+
+test('execute a typescript file with imports with no extensions', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-import-no-extension.ts'),
+  ]);
+
+  match(result.stderr, /Error \[ERR_MODULE_NOT_FOUND\]:/);
+  strictEqual(result.stdout, '');
+  strictEqual(result.code, 1);
+});
+
+test('execute a typescript file with enums', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-enums.ts'),
+  ]);
+
+  // This error should be thrown during transformation
+  match(result.stderr, /SyntaxError: Unexpected reserved word/);
+  strictEqual(result.stdout, '');
+  strictEqual(result.code, 1);
+});
+
+test('execute a typescript file with experimental decorators', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--experimental-typescript',
+    fixtures.path('typescript/test-experimental-decorators.ts'),
+  ]);
+  // This error should be thrown during transformation
+  match(result.stderr, /Expression expected/);
```

Comment:
@kdy1 it would make more sense to throw a specific error during transformation otherwise the syntax error is not very helpful

Reply:
Thats exactly whats needed for typescript features that will not be transpiles